cmd/update-controller/main.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/coreos/bootkube/pkg/cluster"
+	"github.com/coreos/bootkube/pkg/cluster/components"
+	"github.com/golang/glog"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/client/cache"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_3"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+func main() {
+	config, err := restclient.InClusterConfig()
+	if err != nil {
+		glog.Fatal(err)
+	}
+	run(clientset.NewForConfigOrDie(config))
+}
+
+func run(client clientset.Interface) {
+	glog.Info("update controller running")
+	versionChan := make(chan *components.Version)
+	handleConfigChange := func(newConfigMap *v1.ConfigMap) {
```

Comment:
Per in-person discussion, we should change the version configuration to a thirdPartyResource